### DISCUSSION TITLE
select all but

### DISCUSSION THREAD

**Discussion by Randalix** [helix-editor/helix#7342] [Q&A] [answered]
I have a file and want to delete all lines that do not contain a keyword.
I've tried solutions suggested here [stackoverflow](https://stackoverflow.com/questions/2078915/a-regular-expression-to-exclude-a-word-string) (negative look-ahead), but I get ```error: look-around, including look-ahead and look-behind, is not supported```.
Is there another way to do this? I can use sed, but in helix?

**Comment by Randalix** [comment]
Ah, I've found a way:

- Select the lines I want to keep
- yank them to the system clipboard
- delete everything
- paste from the system clipboard
- delete extra lines

Let me know if there is a better way.

**Comment by mweihs** [comment]
Let's say you want to keep all lines with "foo". Select whole file ("%"), pipe to grep:
    `| grep -v foo`

**Comment by pascalkuthe** [comment, marked as answer]
You can use `A-K` to remove selections that match a regex so `%A-sA-K<word><ret>d` does what you want (select entire file, split selection into lines, remove selections that contain regex/word, delete

**Reply by Randalix** [reply]
Nice, thanks!

**Comment by hholst80** [comment]
I am using Helix 24.7 and ALT-K no longer seems to do anything?
